src/storage/config.rs

```diff
@@ -30,20 +33,48 @@ pub enum StorageConfig {
     InMemory,
 }
 
-impl StorageConfig {
-    pub fn into_storage(self) -> Result<Box<dyn Storage>> {
-        Ok(match self {
+#[cfg(feature = "storage-sqlite")]
+impl TryFrom<StorageConfig> for SqliteStorage {
```

Comment:
Relating to my comment in `src/storage/mod.rs`, this is pretty awkward: this is taking an enum type and returning a concrete type _if_ the correct enum variant is present, otherwise producing a runtime error. This should be a build-time error, and I think just instantiating the `{Sqlite,InMemory}Storage` types directly as arguments to `Replica::new` is the simplest and best way to accomplish that.

It means the `StorageConfig` type and this module can go away entirely!